src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Macros/BaseGeneratedSymbolMacro.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Core.Contracts;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Abstractions;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Macros.Config;
+
+namespace Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Macros
+{
+    internal abstract class BaseGeneratedSymbolMacro<T> : BaseMacro<T>, IGeneratedSymbolMacro<T> where T : BaseMacroConfig, IMacroConfig
```

Comment:
Why `IMacroConfig` and not `IGeneratedSymbolConfig`?
Seems like other types lead to runtime error anyways

Reply:
same as https://github.com/dotnet/templating/pull/5223#discussion_r986818889
`T` is not `IGeneratedSymbolConfig`, `T` should be created from `IGeneratedSymbolConfig`.